src/bot/commands/report.ts

```diff
@@ -0,0 +1,303 @@
+        filter: (i: MessageComponentInteraction) => i.user.id === interaction.user.id,
+        time: 60_000
+      });
+      collector.once('end', async (interactions, reason) => {
+        const buttonInteraction = interactions.first();
+        if (!buttonInteraction) {
+          if (reason !== 'time') resolve(null);
+          if (message.deletable) await message.delete();
+          await interaction.reply(interactionInfo('Les 60 secondes se sont ecoulées.'));
+          return resolve(null);
+        }
+
+        return resolve(buttonInteraction);
+      });
+    });
+  }
+
+  async getDoublon(
+    interaction: ChatInputCommandInteraction<'cached'>,
+    joke: Joke
+  ): Promise<JokeCorrectionPayload | null> {
+    const bestDoublons: Joke[] = Jokes.list.filter((j) => j.id < 11);
+    for (const b of Jokes.list) {
+      const match = compareTwoStrings(
+        `${joke.joke.toLowerCase()} ${joke.answer.toLowerCase()}`,
+        `${b.joke.toLowerCase()} ${b.answer.toLowerCase()}`
+      );
+      for (const _b of bestDoublons) {
+        const _match = compareTwoStrings(
+          `${joke.joke.toLowerCase()} ${joke.answer.toLowerCase()}`,
+          `${_b.joke.toLowerCase()} ${_b.answer.toLowerCase()}`
+        );
+        if (match > _match) {
+          const index = bestDoublons.indexOf(_b);
+          bestDoublons.splice(index, 1);
+          bestDoublons.push(b);
+          bestDoublons.sort((d1: Joke, d2: Joke) => {
+            const match = compareTwoStrings(
+            `${joke.joke.toLowerCase()} ${joke.answer.toLowerCase()}`,
+            `${d1.joke.toLowerCase()} ${d1.answer.toLowerCase()}`
+            );
+            const _match = compareTwoStrings(
+            `${joke.joke.toLowerCase()} ${joke.answer.toLowerCase()}`,
+            `${d2.joke.toLowerCase()} ${d2.answer.toLowerCase()}`
+            );
+            return (match > _match) ? 1 : -1
+            }
+          )
+        }
+      }
+    }
+    const question = await interaction.reply({
+      embeds: [
+        {
+          title: `Quel est le doublon de la blague suivante ? (ID : \`${joke.id}\`)`,
+          description: `
+          > **Type**: ${CategoriesRefsFull[joke.type]}
+          > **Blague**: ${joke.joke}
+          > **Réponse**: ${joke.answer}
+
+          Répondez par \`cancel\` pour annuler la commande.
+          `,
+          color: Colors.PROPOSED
+        }
+      ],
+      components: [
+        {
+          type: ComponentType.ActionRow,
+          components: [
+            {
+              type: ComponentType.SelectMenu,
+              customId: 'doublons',
+              placeholder: 'Choisissez un doublon...',
+              // TODO : Il faudrait faire quelque chose avec bestDoublons, mais j'y arrive pas... Au secours
+              options: Object.entries({texte_hey: 'Hey !'}).map(([key, name]) => ({
+                label: name,
+                value: key,
+                description: 'JokeTypesDescriptions[key as Category]'
```

Comment:
Sa serait peut etre mieux comme sa nan ?
```suggestion
    const { ratings } = findBestMatch(
      `${joke.joke} ${joke.answer}`,
      Jokes.list.map((entry) => `${entry.joke} ${entry.answer}`)
    );

    const question = await interaction.reply({
      embeds: [
        {
          title: `Quel est le doublon de la blague suivante ? (ID : \`${joke.id}\`)`,
          description: `
          > **Type**: ${CategoriesRefsFull[joke.type]}
          > **Blague**: ${joke.joke}
          > **Réponse**: ${joke.answer}

          Répondez par \`cancel\` pour annuler la commande.
          `,
          color: Colors.PROPOSED
        }
      ],
      components: [
        {
          type: ComponentType.ActionRow,
          components: [
            {
              type: ComponentType.SelectMenu,
              customId: 'doublons',
              placeholder: 'Choisissez un doublon...',
              // TODO : Il faudrait faire quelque chose avec bestDoublons, mais j'y arrive pas... Au secours
              options: Object.entries(ratings).map(([key, joke]) => ({
                label: joke.target,
                value: key
```

Reply:
Est-ce que ça prend bien les 10 blagues les plus ressemblantes ?

Reply:
Sa recupere tout il te suffira de les trier par niveaux de resemblance et combien tu en veut